lib/datadog/appsec/response.rb

```diff
@@ -43,20 +43,22 @@ def negotiate(env)
         private
 
         FORMAT_MAP = {
-          'text/plain' => :text,
-          'text/html' => :html,
           'application/json' => :json,
+          'text/html' => :html,
+          'text/plain' => :text,
         }.freeze
 
-        DEFAULT_CONTENT_TYPE = 'text/plain'
+        DEFAULT_CONTENT_TYPE = 'application/json'
 
         def content_type(env)
           return DEFAULT_CONTENT_TYPE unless env.key?('HTTP_ACCEPT')
 
-          accepted = env['HTTP_ACCEPT'].split(',').map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
+          accept_types = env['HTTP_ACCEPT'].split(',').map(&:strip)
+
+          accepted = accept_types.map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
 
-          accepted.each_with_object(DEFAULT_CONTENT_TYPE) do |range, _default|
-            match = FORMAT_MAP.keys.find { |type| range === type }
+          accepted.each do |range|
+            match = FORMAT_MAP.keys.find { |key| range === key }
```

Comment:
`FORMAT_MAP` keys are content types, so I'm questioning the name change here.

I would even change it that way to make it clearer:

```
type_match = FORMAT_MAP.keys.find { |type| range === type }

return type_match if type_match
```